src/components/DAO/ExpandDAO/DeploySeele/index.tsx

```diff
@@ -61,47 +56,58 @@ const DeploySeele: FunctionComponent<{
 				proposal => proposal.type === "decentralizeDAO" && proposal.state === "active"
 			)
 			if (expandProposal) {
-				setStage("waiting")
+				// TODO: Redirect to proposal details page
+				console.log("TODO")
 			}
 		}
 	}, [proposals])
 
+	const handleProceedToConfirm = async () => {
+		if (signer) {
```

Comment:
You mess up the whole data flow and make it ambiguous with this edition.
What if we go back from confirm stage?
You add the multiTx as a regular tx in the sequence, and then search for it.
I see that it's handled differently, but with this approach we can't always be sure we have this tx in right place.
There's an effect above, where we build the transaction sequence. I suggest you to create a new state for this multiTx and build it that effect each time when we add or remove a strategy (and include the case when we remove all strategies and multiTx becomes null)
Than we'll always be sure that we have it in place and it's always up-to-date

Reply:
Partially changed - I don't see it necessary to re-build `multiTx` each time when we add/remove strategies, but rebuilding it each time when we clicking "Continue" button - thus we doing it only when user is kind of sure about content of multiTx

Reply:
First, don't set state with result of await call. I've even done a small research to figure out what this will lead to.
If you add a considerable delay to `buildMultiSendTx` function, the further `handleProceedToConfirm` function execution will freeze until this promise resolves. So you potentially freeze the UI until it is executed.
Second, this approach, again, messes up the data flow. `MultiTx` is a derivative of tx sequence and nothing else. You make it depend on the `stage`, which is truly redundant.
Please, get back to the approach I mentioned. You can write an async function to avoid chaining and call it in effect. First, build tx sequence, then, build multiTx, then, update the state. With this approach you can be sure the data to be always up-to-date.
And FYI, `buildMultiSendTx` is off-chain, so it's not a problem to update it with each tx sequence change.

Reply:
> First, don't set state with result of await call.

Technically - there's no difference. We either put value to the variable or directly put that to the state. And UI might potentially freeze anyway, but the difference will be that it could happen after each voting strategy selection/removal, or only once - before proceeding to the confirmation. And the last one makes much more sense to me - we don't need "intermediate" multiTx, but the only final one. We don't need multiTx generated before actually going to confirmation without a complete set of transactions, but only with the final batch.

I would say we need 3rd opinion here.

Reply:
multitx really is more of a final stage action, re-building it everytime a new transaction is added or removed from the multitx bundle seems a bit redundant. I'm going to merge this in the interest of time